piv/key.go

```diff
@@ -764,7 +785,15 @@ func (k KeyAuth) authTx(yk *YubiKey, pp PINPolicy) error {
 	// PINPolicyAlways should always prompt a PIN even if the key says that
 	// login isn't needed.
 	// https://github.com/go-piv/piv-go/issues/49
-	if pp != PINPolicyAlways && !ykLoginNeeded(yk.tx) {
+
+	var flag bool
+
+	yk.withTx(func(tx *scTx) error {
```

Comment:
This should probably only be called once per method call? Maybe change authTx to take a transaction then only call withTx from exported methods?